### PR TITLE
fix: store v3 misc bugs

### PR DESCRIPTION
**Pull request by SionoiS**
Various bug fixes.

For https://github.com/waku-org/nwaku/issues/2717
I just remove the `assert` and the code handle the error just fine.

For https://github.com/waku-org/nwaku/issues/2715
I turn off the plus sign for url de/encoding and change to base 64 for json.

For https://github.com/waku-org/nwaku/issues/2716
I added a cursor check before the main query.

For https://github.com/waku-org/nwaku/issues/2719
I change the serde for REST api

### CODE REVIEW

File: tests/waku_archive/test_driver_postgres_query.nim [hunk 1]
```diff
@@ -645,6 +645,50 @@ suite "Postgres driver - queries":
...
+        fakeWakuMessage(@[byte 0], ts = ts(00)),
+        fakeWakuMessage(@[byte 1], ts = ts(10)),
+        fakeWakuMessage(@[byte 2], contentTopic = contentTopic, ts = ts(20)),
+        fakeWakuMessage(@[byte 3], contentTopic = contentTopic, ts = ts(30)),
+        fakeWakuMessage(@[byte 4], contentTopic = contentTopic, ts = ts(40)),
+        fakeWakuMessage(@[byte 5], contentTopic = contentTopic, ts = ts(50)),
+        fakeWakuMessage(@[byte 6], contentTopic = contentTopic, ts = ts(60)),
+        fakeWakuMessage(@[byte 7], contentTopic = contentTopic, ts = ts(70)),
+      ]
+    var messages = expected
```
Comment (gabrielmer) [inline tests/waku_archive/test_driver_postgres_query.nim:663]: We don't really use the `expected` variable here, so we can get rid of it (same in the other tests)

```suggestion
    let messages =
      @[
        fakeWakuMessage(@[byte 0], ts = ts(00)),
        fakeWakuMessage(@[byte 1], ts = ts(10)),
        fakeWakuMessage(@[byte 2], contentTopic = contentTopic, ts = ts(20)),
        fakeWakuMessage(@[byte 3], contentTopic = contentTopic, ts = ts(30)),
        fakeWakuMessage(@[byte 4], contentTopic = contentTopic, ts = ts(40)),
        fakeWakuMessage(@[byte 5], contentTopic = contentTopic, ts = ts(50)),
        fakeWakuMessage(@[byte 6], contentTopic = contentTopic, ts = ts(60)),
        fakeWakuMessage(@[byte 7], contentTopic = contentTopic, ts = ts(70)),
      ]
```

**Review by gabrielmer — APPROVED** [review]
LGTM, thank you!

File: tests/waku_archive/test_driver_postgres_query.nim (hunk 1 above)
Comment (SionoiS) [reply tests/waku_archive/test_driver_postgres_query.nim:663]: yes! sorry for the copy pasta

**Review by NagyZoltanPeter — APPROVED** [review]
Nice bunch of fixes, thank you!

**Comment by SionoiS** [comment]
Ran all the related tests locally. Will merge even if CI is broken. :crossed_fingers:

**Pull request merged**